### PR TITLE
🛂 Implement OAuth2

### PR DESCRIPTION
**Pull request by DRSchlaubi**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -41,6 +41,11 @@ android {
         release {
             minifyEnabled false
             proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
+            buildConfigField "android.net.Uri", "OAUTH_SERVER", "android.net.Uri.parse(\"https://api.traewelldroid.de\")"
```
Comment (jheubuch) [inline app/build.gradle:44]:
```suggestion
            buildConfigField "android.net.Uri", "OAUTH_SERVER", "android.net.Uri.parse(\"https://app.traewelldroid.de\")"
```
I think that should be the callback url?

File: app/src/main/res/drawable/ic_popup.xml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="24dp"
+    android:height="24dp"
+    android:autoMirrored="true"
+    android:tint="#000000"
```
Comment (jheubuch) [inline app/src/main/res/drawable/ic_popup.xml:5]:
```suggestion
    android:tint="?attr/colorControlNormal"
```

**Review by jheubuch — CHANGES_REQUESTED** [review]

File: app/build.gradle (hunk 1 above)
Comment (DRSchlaubi) [reply app/build.gradle]: I am still not fully convinced that it is a great Idea to have the server run on the deeplink URL, but ig it works

File: app/src/main/res/drawable/ic_popup.xml (hunk 2 above)
Comment (jheubuch) [reply app/src/main/res/drawable/ic_popup.xml:5]: It looked pretty weird on light theme, I'll have a try...

Comment (jheubuch) [reply app/src/main/res/drawable/ic_popup.xml:5]: Gotcha!

Comment (DRSchlaubi) [reply app/src/main/res/drawable/ic_popup.xml]: Looks good

**Review by jheubuch — APPROVED** [review]

**Review by jheubuch — APPROVED** [review]

**Pull request merged**